rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -760,51 +782,53 @@ void main_window::InstallPup(QString file_path)
 			QMessageBox::Yes | QMessageBox::No, QMessageBox::No) != QMessageBox::Yes)
 		{
 			gui_log.notice("Firmware: Cancelled installation from drop. File: %s", sstr(file_path));
-			return;
+			return false;
 		}
 	}
 
-	if (!file_path.isEmpty())
-	{
-		// Handle the actual installation with a timeout. Otherwise the source explorer instance is not usable during the following file processing.
-		QTimer::singleShot(0, [this, file_path]()
-		{
-			HandlePupInstallation(file_path);
-		});
-	}
+	return HandlePupInstallation(this, file_path);
 }
 
-void main_window::HandlePupInstallation(QString file_path)
+bool main_window::HandlePupInstallation(main_window* mw, QString file_path)
 {
 	if (file_path.isEmpty())
 	{
-		return;
+		return false;
 	}
 
-	if (!m_gui_settings->GetBootConfirmation(this))
+	if (mw && !m_gui_settings->GetBootConfirmation(mw))
 	{
-		return;
+		return false;
 	}
 
 	Emu.SetForceBoot(true);
 	Emu.Stop();
 
-	m_gui_settings->SetValue(gui::fd_install_pup, QFileInfo(file_path).path());
 	const std::string path = sstr(file_path);
 
+	const auto critical = [mw](QString str)
```

Comment:
this can be ommited, as this function cannot and shouldn't be called without a main window.

Reply:
Removed.